### ISSUE TITLE
Tuya Outlets not working

### ISSUE THREAD

**Issue by Casier98** [rospogrigio/localtuya#371]
I purchased a few of these Tuya-based smart outlets a while ago:
https://www.costco.ca/ce-smart-home-wi-fi-smart-plug-with-energy-monitoring,-2-pack.product.100476838.html

And recently, I've been trying to get them to work with Home Assistant, without much luck.  I've tried adding them manually to the configuration.yaml file as so:
![Capture2](https://user-images.githubusercontent.com/7659117/109251975-7249e580-77ba-11eb-9c81-0ea07a073e89.PNG)

(I've tried both 3.1 and 3.3 for protocol_version, no difference)

And all I get is the light itself as unavailable:
![Capture](https://user-images.githubusercontent.com/7659117/109252026-8c83c380-77ba-11eb-8dc9-39541b351ba2.PNG)

I've also tried having my configuration.yaml file cleared of any localtuya devices and tried adding them through the UI, but seem to get an unknown error: 
![image](https://user-images.githubusercontent.com/7659117/109252219-fbf9b300-77ba-11eb-8c6b-9ad826a9b60b.png)

The logs show this:
![image](https://user-images.githubusercontent.com/7659117/109252326-454a0280-77bb-11eb-85ef-f21102df1ea4.png)

(Ignore the discovered device...it is another Tuya device on my network that I'm not setting up at the moment).

I was able to validate that the keys for the devices are correct, as I just pulled them, and also had them working in a HomeBridge set up earlier (but obviously don't want to keep them there).

Any insights will be appreciated.

**Comment by Akshue** [comment]
Have you tried ID 1?  Switches are normally that.

**Comment by Casier98** [comment]
Yeah, I did...the id 2 came from me doing a lot of experimentation on what might be wrong.  I initially had like 5 devices set up, but when they weren't working, I stuck with just 1 for the time being until I get things sorted.

**Comment by stratus-ss** [comment]
I'm experiencing the same symptoms although my devices are on a different VLAN. HA has access to this VLAN and can control the TASMOTA/ESPHome devices on this VLAN just fine. I used the values from the description in the README:
```
  - host: 192.168.66.82
    device_id: 14xxxxxxd514e
    local_key: 1cbxxxxxx010
    friendly_name: Tuya Device
    protocol_version: "3.3"
    entities:
      - platform: switch
        friendly_name: Plug
        id: 1
```
